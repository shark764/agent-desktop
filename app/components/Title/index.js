/**
*
* Title
*
*/

import React, { PropTypes } from 'react';
import { FormattedMessage } from 'react-intl';
import Radium from 'radium';

function Title(props) {
  const styles = {
    base: {
      fontFamily: 'ProximaNova',
      fontSize: '20px',
      fontWeight: 'bold',
      fontStyle: 'normal',
      fontStretch: 'normal',
      color: '#494949',
    },
  };

  return (
    <span style={[styles.base, props.style]}>
      <FormattedMessage style={styles.center} {...props.text} />
    </span>
  );
}

Title.propTypes = {
  text: PropTypes.object.isRequired,
  style: PropTypes.object,
};

export default Radium(Title);
