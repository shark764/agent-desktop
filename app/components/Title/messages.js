/*
 * Title Messages
 *
 * This contains all the text for the Title component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'app.components.Title.header',
    defaultMessage: 'This is the Title component !',
  },
});
