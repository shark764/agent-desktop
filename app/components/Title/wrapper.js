import styled from 'styled-components';

const Wrapper = styled.div`
    font-family: ProximaNova;
    font-size: 20px;
    font-weight: bold;
    font-style: normal;
    font-stretch: normal;
    color: #494949;
`;

export default Wrapper;

