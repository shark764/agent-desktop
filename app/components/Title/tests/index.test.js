/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import Title from '../index';

describe('<Title />', () => {
  it('should render correctly', () => {
    const rendered = shallow(
      <Title
        id="mockId"
        text={{ id: 'app.containers.Login.welcome' }}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
