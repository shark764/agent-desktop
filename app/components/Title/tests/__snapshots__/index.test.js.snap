// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render correctly 1`] = `
<span
  data-radium={true}
  id="mockId"
  style={
    Object {
      "color": "#494949",
      "fontSize": "20px",
      "fontStretch": "normal",
      "fontStyle": "normal",
      "fontWeight": "bold",
    }
  }
>
  <FormattedMessage
    id="app.containers.Login.welcome"
    values={Object {}}
  />
</span>
`;
